javascript/ql/src/experimental/Security/CWE-079/ClipboardXss.ql

```diff
@@ -0,0 +1,78 @@
+/**
+ * @name Client-side clipboard-based cross-site scripting
+ * @description Pasting clipboard input directly to the DOM without proper sanitization allows for
+ *              a cross-site scripting vulnerability.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 3.1
+ * @precision high
+ * @id js/clipboard-xss
+ * @tags security
+ *       external/cwe/cwe-079
+ */
+
+import javascript
+import DataFlow
+import semmle.javascript.security.dataflow.DomBasedXssQuery as DomBasedXss
+import DataFlow::PathGraph
+
+/*
+ * Gets references to clipboardData DataTransfer objects
+ * https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData
+ */
+
+SourceNode clipboardDataTransferSource(TypeTracker t) {
+  t.start() and
+  exists(DataFlow::PropRead pr | pr.getPropertyName() = "clipboardData" and pr.flowsTo(result))
+  or
+  exists(TypeTracker t2 | result = clipboardDataSource(t2).track(t2, t))
```

Comment:
```suggestion
  exists(TypeTracker t2 | result = clipboardDataTransferSource(t2).track(t2, t))
```
The missed result you mentioned in the related issue was caused by this typo.

Reply:
Thanks! This was probably causing me other problems I was not able to diagnose.